app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -30,6 +30,8 @@ def validate_config_info(options)
           vm_options = {}
           vm_options[:pre_ansible_playbook_service_template_id] = pre_service_id if vm_hash[:pre_service]
           vm_options[:post_ansible_playbook_service_template_id] = post_service_id if vm_hash[:post_service]
+          vm_options[:osp_security_group_id] = vm_hash[:osp_security_group] if vm_hash[:osp_security_group].present?
```

Comment:
The backend accepts either `vm_hash[:osp_security_group]` as an object or `vm_hash[:osp_security_group_id]` as an id. You can convert them all to ids and save in `vm_options[:osp_security_group_id]`. 
Please look at line 25 and 27 as an example. Either `vm_hash[:vm_id]` or `vm_hash[:vm]` is accepted.

Same for `osp_flavor_id`.

Reply:
The `actions` hash coming from the frontend looks like this -
```
"actions": [
  {
    "vm_id": "20000000000057",
    "pre_service": false,
    "post_service": false,
    "osp_security_group": "20000000000007",
    "osp_flavor": "20000000000001"
  }
]
```

So the security_group and flavor info is available here in terms of `id` and not `object`.

On the frontend, we should perhaps change the `osp_` keys in the `actions` hash to be `osp_security_group_id` and `osp_flavor_id`, just to be consistent with the naming convention.

Reply:
In your example, yes, please change the frontend to use group and flavor ids.

Although v2v frontend never sends actual object, our backend for all service templates always accept both object and ids.